addons/hr_timesheet/models/project.py

```diff
@@ -157,12 +157,12 @@ def _compute_total_hours_spent(self):
 
     @api.depends('child_ids.effective_hours', 'child_ids.subtask_effective_hours')
     def _compute_subtask_effective_hours(self):
-        for task in self:
+        for task in self.with_context(active_test=False):
             task.subtask_effective_hours = sum(child_task.effective_hours + child_task.subtask_effective_hours for child_task in task.child_ids)
 
     def action_view_subtask_timesheet(self):
         self.ensure_one()
-        tasks = self._get_all_subtasks()
+        tasks = self._get_all_subtasks(include_archived=True)
```

Comment:
Why did not you use the `self.with_context(active_test=False)` rather than to change the signature of this method ?

```suggestion
        tasks = self.with_context(active_test=False)._get_all_subtasks()
```